### DISCUSSION TITLE
Enable UART shell issue

### DISCUSSION THREAD

**Discussion by diefish2** [zephyrproject-rtos/zephyr#55058] [Q&A]
I am trying to enable the shell on zephyr OS; however, I only see the printf but do not see the prompt pop-up. I am wondering that what did I need to check in order to fix the issue. 

- Currently i am using zephyr@6aa783ed6a which is version 2.7.99
- uart_ns16550 is my uart driver
- on my prj.conf file, I have the following configuration as well
1. CONFIG_CONSOLE=y
2. CONFIG_UART_CONSOLE=y
3. CONFIG_UART_NS16550=y
4. CONFIG_SHELL=y
5. CONFIG_SERIAL=y
- on my .dts file, I have the following setup:
/ {
	chosen {
		zephyr,console = &uart0;
		zephyr,shell-uart = &uart0;
		zephyr,sram = &ram0;
	};
};

&uart0 {
	status = "okay";
	 current-speed = <115200>;
        clock-frequency = <32500000>;
};

- on my main.c; I already have the following code which create the shell command: 

  SHELL_STATIC_SUBCMD_SET_CREATE(sub_demo,
	  SHELL_CMD(dictionary, &sub_dict_cmds, "Dictionary commands", NULL),
	  SHELL_CMD(hexdump, NULL, "Hexdump params command.", cmd_demo_hexdump),
	  SHELL_CMD(params, NULL, "Print params command.", cmd_demo_params),
	  SHELL_CMD(ping, NULL, "Ping command.", cmd_demo_ping),
	  SHELL_CMD(board, NULL, "Show board name command.", cmd_demo_board),
  #if defined CONFIG_SHELL_GETOPT
	  SHELL_CMD(getopt, NULL,	"Cammand using getopt, looking for: \"abhc:\".",
		    cmd_demo_getopt),
  #endif
	  SHELL_SUBCMD_SET_END /* Array terminated. */
  );

Therefore I am wondering that what do I need to check in order to see the prompt on my UART.

Thank you

**Comment by billkarduck** [comment]
First make sure your device tree overlay file has selected the correct  RX and TX pins selected. Mine looks like this:
&uart0 {
			current-speed = <115200>;
			status = "okay";
			pinctrl-0 = <&uart0_default>;
			pinctrl-names = "default", "sleep";
		};

&pinctrl {
	uart0_default: uart0_default {
		group1 {
			psels = <NRF_PSEL(UART_TX, 0, 6)>,
				<NRF_PSEL_DISCONNECTED(UART_RTS)>;
		};
		group2 {
			psels = <NRF_PSEL(UART_RX, 0, 8)>,
				<NRF_PSEL_DISCONNECTED(UART_CTS)>;
				bias-pull-up;
		};
	};
};

However, I too have an issue where I get the printk() function output to my console, but no shell interface is functioning. I have the same code working on a very old(3 years ago) release of Noedic Nrf Connect and Zephyr. But porting to the latest release my SHELL does not work either. First check your device tree and test with printk() or similar. Hopefully I can figure out what is wrong or someone else can help both of us.

**Reply by billkarduck** [reply]
Figured out my issue. Had an interrupt priority issue. Everything works for me now.

**Comment by joakim-tjernlund** [comment]
I am in the same position printf is working from threads but whatever I do I cannot enter input. I see the `uart:~$` prompt
but no input can be entered. Any ideas welcome!
